src/modules/Experience/components/Cursor/Cursor.tsx

```diff
@@ -0,0 +1,114 @@
+import { useScroll, Text } from '@react-three/drei';
+import * as THREE from 'three'
+import gsap from 'gsap';
+import { useRef } from 'react';
+import { cursorFragmentShader, cursorVertexShader } from '../../shaders/cursorShader';
+import { fontLibrary, textureLibrary } from '@/helpers';
+import { useFrame, useThree } from '@react-three/fiber';
+import { useCursor } from './CursorManager';
+import { theme } from '@/theme/theme';
+import { insideCursorFragmentShader, insideCursorVertexShader } from '../../shaders/cursorShader/insideCursorShader';
+
+const Cursor = () => {
+  const { settings } = useCursor()
+  const { viewport } = useThree()
+  const scroll = useScroll();
+
+  const ref = useRef<THREE.Mesh | any>()
+  const insideRef = useRef<THREE.Mesh | any>()
```

Comment:
Would you remember ewhat does ref and insideRef are in 1 year? XD maybe be a better name, is there a reason they still use ANY?  We can solve this together maybe today?